includes/Modules/Analytics_4.php

```diff
@@ -882,6 +904,61 @@ public function check_service_entity_access() {
 		return true;
 	}
 
+	/**
+	 * Gets the Google Tag Settings for the given measurement ID.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Google_Service_TagManager_Container $container Tag Manager container.
+	 * @param string                              $measurement_id Measurement ID.
+	 * @return array Google Tag Settings.
+	 */
+	protected function get_google_tag_settings_for_measurement_id( $container, $measurement_id ) {
```

Comment:
Initially I did that but it breaks our convention of `create_data_request` -> `parse_data_request`. I faced problems while testing the route in phpunit. The `create_data_request`  was passing the unprocessed `Request` object to the function (most likely because it was being deferred). 

Also this approach is more "correct" as we only manipulate or augment API data in the `parse_data_request` function for all other API. 

Cheers.

Reply:
Thanks for the explanation here, the approach you've taken SGTM :+1: